models/model_choice.py

```diff
@@ -137,9 +138,10 @@ def set_hyperparameters(params,
     gamma = get_key_def('gamma', params['scheduler']['params'], 0.9)
     class_weights = torch.tensor(class_weights) if class_weights else None
     # Loss function
-    criterion = MultiClassCriterion(loss_type=loss_fn,
-                                    ignore_index=dontcare_val,
-                                    weight=class_weights)
+    if loss_fn['_target_'] == 'torch.nn.CrossEntropyLoss':
+        criterion = instantiate(loss_fn, weight=class_weights)  # FIXME: unable to pass this through hydra
```

Comment:
I really don't like this. CrossEntropy loss (multiclass) is the only loss that uses class_weights as a parameter. We had adapted all other losses to accept it but no use it. The binary losses that @victorlazio109 has added from smp don't accept the "weight" parameter and therefore raise an exception if this parameter is passed.

Logically, class_weights would have been in the crossentropy.yaml config directly (and not in model_choice). It could therefore be dealt with directly by hydra, but one things makes it a bit tricky:
Cross entropy in torch.nn [only accepts a tensor](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/loss.py#L1095). If it could handle a list, this would be much more convenient for hydra.

The only way hydra can directly input a tensor to the weight argument is by changing the CE yaml to somtehing like this:
# @package _global_
loss:
```
    _target_: torch.nn.CrossEntropyLoss
    ignore_index: ${dataset.ignore_index}
    weight:
        _target_: torch.tensor
        data: ${dataset.class_weights}
```
This works fine when class_weights is not None (ex.: [0.5,0.5] successfully gets converted to tensor), but when it's None torch.tensor(None) raises and exception. It would require hydra to have some king of if/else statement if class_weights is None.

Anyways, I'm leaving it to you guys if you have ideas. I spent enough time on this case.